pydantic/types.py

```diff
@@ -1505,16 +1515,16 @@ def get_json_schema(_core_schema: core_schema.CoreSchema, handler: GetJsonSchema
 
         json_schema = core_schema.no_info_after_validator_function(
             source,  # construct the type
-            inner_schema,
+            cls._inner_schema,
         )
         s = core_schema.json_or_python_schema(
             python_schema=core_schema.union_schema(
                 [
                     core_schema.is_instance_schema(source),
                     json_schema,
                 ],
-                strict=True,
-                custom_error_type=error_kind,
+                strict=cls._strict,
```

Comment:
are we no longer respecting strict?

Reply:
i removed the strict hardcoded values, but when doing
```py
StrictSecretDate = Annotated[SecretDate, Strict()]
```

i get this: 
```py
RuntimeError: Unable to apply constraint strict to schema json-or-python
```

i'll continue to search for a solution, but any help is appreciated

Reply:
fixed with lax-or-strict